core/src/heaviest_subtree_fork_choice.rs

```diff
@@ -56,23 +85,55 @@ struct ForkInfo {
     stake_voted_subtree: ForkWeight,
     // Best slot in the subtree rooted at this slot, does not
     // have to be a direct child in `children`
-    best_slot: Slot,
-    parent: Option<Slot>,
-    children: Vec<Slot>,
+    best_slot: SlotHashKey,
+    parent: Option<SlotHashKey>,
+    children: Vec<SlotHashKey>,
     // Whether the fork rooted at this slot is a valid contender
     // for the best fork
     is_candidate: bool,
+    duplicate_votes: HashMap<(Pubkey, Slot), u64>,
```

Comment:
I feel pretty convinced that just picking one of the duplicate votes sorted by blockhash is the way to go. I think of it similarly to how vote lockout violations would be treated. Imagine you have a validator who votes on slot 10 in fork A and slot 11 in fork B. It's a bit arbitrary which fork you choose in the same way that choosing based on blockhash is arbitrary for dupe slot votes.

Reply:
@carllin thoughts here?